src/server.rs

```diff
@@ -72,11 +73,22 @@ where
     }
 }
 
+/// Given a host and port, creates a newstring of the form "host:port" or
+/// "[ho:st]:port", depending on whether the host is an IPv6 literal.
+fn join_host_port(host: &str, port: u16) -> String {
+    let is_ipv6 = host.parse::<Ipv6Addr>().is_ok();
```

Comment:
How about using `SocketAddr` and `Display` instead? So we don't need to know the details.

Reply:
Host many be a domain name.

Reply:
Yes, then fallback to the old way should be OK.